### ISSUE TITLE
UHK Agent 4.2.1 on macOS Monterey 12.7.6 triggers a crash report when being closed

### ISSUE THREAD

**Issue by jbm1** [UltimateHackingKeyboard/agent#2354]
Please find attached a log of macOS crash reporting mechanism

[2024-10-15-UHK-Agent-4.2.1-on-macOS-12-7-6-crashes-at-exit.txt.txt](https://github.com/user-attachments/files/17381043/2024-10-15-UHK-Agent-4.2.1-on-macOS-12-7-6-crashes-at-exit.txt.txt)

![Screenshot 2024-10-15 at 15 32 54](https://github.com/user-attachments/assets/52d6b41c-2543-4bb2-807c-0361af1a9394)

_Agent version: 4.2.1_
_Right keyboard half firmware: UltimateHackingKeyboard/firmware: 11.2.0
 Left keyboard half firmware: UltimateHackingKeyboard/firmware: 11.2.0
 Key cluster firmware: UltimateHackingKeyboard/firmware: 11.2.0
 Trackpoint firmware: UltimateHackingKeyboard/firmware: 11.2.0_

3. Our contact form will suggest you solutions, so copy-paste your message at https://ultimatehackingkeyboard.com/contact-us - 

_[Where can I download Agent?](https://ultimatehackingkeyboard.com/knowledgebase/where-can-i-download-agent)
If you have a UHK, make sure to complete the [onboarding guide](https://ultimatehackingkeyboard.com/manuals/uhk60v2) before downloading Agent (the configurator of the UHK) to familiarize yourself with the features of your UHK. On the last page of the onboarding guide, you can [download Agent](https://ultimatehackingkeyboard.com/agent).

If you don't have a UHK, try [the web demo of Agent](http://ultimatehackingkeyboard.github.io/agent/).

[Do you support Linux, Mac, and Windows?](https://ultimatehackingkeyboard.com/knowledgebase/do-you-support-linux-mac-and-windows)
Yes. [Agent](https://ultimatehackingkeyboard.com/knowledgebase/where-can-i-download-agent), the UHK configurator, runs on Linux, Mac, and Windows 10+._

**Comment by ert78gb** [comment]
Unfortunately, it happens sometimes. Looks like Electron does not handle properly the exit process on Mac. We are waiting for the upstream bug fixing.

**Comment by mondalaci** [comment]
@ert78gb Is this still an issue? If so, please link the upstream bug.

**Comment by ert78gb** [comment]
Yes, still an issue. https://github.com/node-hid/node-hid/issues/554

I will improve Agent to decrease the chance of the error, but I can't eliminate it 100% without electron and/or node-hid.

**Comment by laxu** [comment]
Can confirm this happens on 8.0.1, MacOS Sequia 15.7.2. Crashes every time I quit it.

Crash log:

[UHK crash log.txt](https://github.com/user-attachments/files/24134367/UHK.crash.log.txt)

**Comment by ert78gb** [comment]
Looks like it was node-hid error. I did not experience the issue after we merged the #3015

I keep the PR open until we release the new version and we will got user feedback

**Comment by pcooke9** [comment]
I'm not sure if it's related or not, but I recently saw similar behavior in Win 11 OS.
The following error was happening (almost) every time I closed Agent, and there were Windows event log entries going all the way back to at least early April. Windows didn't show any error popups etc., but the cursor with hour glass briefly appeared when it happened.

A recent Agent update has resolved it for me as well. 


<details>
<summary>Win 11 Event Log</summary>

```
- System 

  - Provider 

   [ Name]  Application Error 
   [ Guid]  {a0e9b465-b939-57d7-b27d-95d8e925ff57} 
 
   EventID 1000 
 
   Version 0 
 
   Level 2 
 
   Task 100 
 
   Opcode 0 
 
   Keywords 0x8000000000000000 
 
  - TimeCreated 

   [ SystemTime]  2026-07-24T20:17:16.4204244Z 
 
   EventRecordID 367807 
 
   Correlation 
 
  - Execution 

   [ ProcessID]  47532 
   [ ThreadID]  41472 
 
   Channel Application 
 
   Computer 
 
  - Security 

   [ UserID]  
 

- EventData 

  AppName UHK Agent.exe 
  AppVersion 10.1.0.0 
  AppTimeStamp 6a2796a7 
  ModuleName UHK Agent.exe 
  ModuleVersion 10.1.0.0 
  ModuleTimeStamp 6a2796a7 
  ExceptionCode c0000409 
  FaultingOffset 0000000002f064a5 
  ProcessId 0x8b54 
  ProcessCreationTime 0x1dd1ba50af2ee97 
  AppPath C:\Program Files\UHK Agent\UHK Agent.exe 
  ModulePath C:\Program Files\UHK Agent\UHK Agent.exe 
  IntegratorReportId b418f573-4e59-4f62-a40f-a78aab240b2a 
  PackageFullName  
  PackageRelativeAppId  

```

</details>

**Comment by ert78gb** [comment]
node-hid fixed many things in the latest release https://github.com/node-hid/node-hid/releases/tag/v3.4.0 around these crashing. Windows is not my daily driver and I did not notice the hour glass. but I am happy if the new update solves windows issues too. Thx for the feedback.

> A recent Agent update has resolved it for me as well.

Just for the clarification , you are referencing to the latest master build and not the latest pinned release as 10.1.0

**Comment by pcooke9** [comment]
> Just for the clarification , you are referencing to the latest master build and not the latest pinned release as 10.1.0

Right, definitely not the official release branch. It looks like it stopped producing errors for me on July, 24th. So I think I was on the last `feature/minimize-to-tray-850` build at the time, which was not long after the node-hid update.

**Comment by mondalaci** [comment]
I can't reproduce this, and according to the reports above, it should be fixed, so I'm closing the issue.
